### ISSUE TITLE
Serializing `Time` and `DateTime`

### ISSUE THREAD

**Issue by tomchristie** [encode/typesystem#40]
Currently unimplemented. See https://github.com/encode/typesystem/blob/master/typesystem/formats.py#L128 and https://github.com/encode/typesystem/blob/master/typesystem/formats.py#L87

Needs:

* Uncommenting out of both those two cases.
* Test coverage.

**Comment by rbw** [comment]
Has anyone started working on this? If not, I can help out.

**Comment by tomchristie** [comment]
That’d be grand, thanks!

**Comment by rbw** [comment]
Is the "zero-offset-to-zulu" conversion in `DateTimeFormat.serialize` necessary?

```python
def serialize(self, obj: typing.Any) -> str:
    value = value.isoformat()
    if value.endswith('+00:00'):
        value = value[:-6] + 'Z'
    return value
```

**Comment by tomchristie** [comment]
Great work!

> Is the "zero-offset-to-zulu" conversion in `DateTimeFormat.serialize` necessary?

Would prefer it, yes.
I think having UTC datetimes always drop the offset is a nice style.
(It's what we use in REST framework for example)

**Comment by rbw** [comment]
I've added the zero-offset to UTC "Z" designator conversion code.

**Comment by florimondmanca** [comment]
For people blocked by this issue while using `orm`, here's a workaround — basically an in-project implementation of #69:

```python
# tspatch.py
import typing

from typesystem import formats
from typesystem.fields import FORMATS


class TimeFormat(formats.TimeFormat):
    def serialize(self, obj: typing.Any) -> str:
        return obj.isoformat() if obj is not None else None


class DateTimeFormat(formats.DateTimeFormat):
    def serialize(self, obj: typing.Any) -> str:
        if obj is None:
            return None
        value = obj.isoformat()
        if value.endswith("+00:00"):
            value = value[:-6] + "Z"
        return value


FORMATS.update({"datetime": DateTimeFormat(), "time": TimeFormat()})
```

`import tspatch` before defining the models then use `orm.DateTime()` and `orm.Time()` as usual.

cc @kcrebound

**Comment by tomchristie** [comment]
Closed via #69
